### PR TITLE
[MNOE-65] / [MNOE-87] - Implemented staff manager page

### PR DESCRIPTION
**Pull request by clemthenem**
### Staff managment
Implemented a staff manager page where an admin can manage the role of a user (Staff or Admin at the moment). Using smart table allows us to sort quickly each column and search for specific info.
Admins can also add or remove a staff.

### Marketplace (customer creation)
Added a title element with a tiny description for each app.
Added a search box to quickly find an app using its name.

### Callbacks
Added Noox module of MnoeObservables. Allow a cleaner use of callbacks and notifiers.

### CODE REVIEW

**Comment by alexnoox** [comment]
@xaun Please review 🍔 

@ouranos FYI, please merge after @xaun's review 🍌

File: frontend-admin/src/app/components/mnoe-api/api.svc.coffee [hunk 1]
```diff
@@ -21,7 +21,10 @@
         # Extract and restangularize the object
         else if (operation == 'get' || operation == 'put' || operation == 'post')
           what = inflector.singularize(what)
-          extractedData = data[what]
+          if (data[what])
+            extractedData = data[what]
+          else
+            extractedData = data
```
Comment (alexnoox) [inline frontend-admin/src/app/components/mnoe-api/api.svc.coffee]: Are you sure that this is used?

File: frontend-admin/src/app/components/mnoe-staffs-list/mnoe-staffs-list.coffee [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+  }
+  controller: ($filter, $log, MnoeUsers, MnoConfirm, MnoeObservables, ADMIN_ROLES, toastr) ->
+    vm = this
+
+    vm.listOfStaff = []
+
+    # Manage sorting, search and pagination
+    vm.callServer = (tableState) ->
+
+      sort   = updateSort (tableState.sort)
```
Comment (alexnoox) [inline frontend-admin/src/app/components/mnoe-staffs-list/mnoe-staffs-list.coffee]: You can remove the space between the function def and the first instruction

File: frontend-admin/src/app/components/mnoe-api/admin/users.svc.coffee [hunk 3]
```diff
@@ -1,24 +1,20 @@
 # Service for managing the users.
-@App.service 'MnoeUsers', ($q, MnoeAdminApiSvc) ->
+@App.service 'MnoeUsers', ($q, $log, MnoeAdminApiSvc, MnoeObservables, ADMIN_ROLES) ->
   _self = @
 
   @list = (limit, offset, sort) ->
     promise = MnoeAdminApiSvc.all('users').getList({order_by: sort, limit: limit, offset: offset}).then(
       (response) ->
-        notifyListObservers(promise)
+        MnoeObservables.notifyObservers('listUserChange', promise)
```
Comment (xaun) [inline frontend-admin/src/app/components/mnoe-api/admin/users.svc.coffee]: index event keys in a constant? prevents magic strings xD

File: frontend-admin/src/app/components/mnoe-staffs-list/mnoe-staffs-list.coffee [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+        setAllStaffsList()
+        return fetchStaffs(vm.staff.nbItems, 0)
+      else if vm.state == 'last'
+        setLastStaffsList()
+        return fetchStaffs(10, 0, 'created_at.desc')
+
+    # Display all the staffs
+    setAllStaffsList = () ->
+      vm.staff.widgetTitle = 'All staffs'
+      vm.staff.switchLinkTitle = '(last 10)'
```
Comment (alexnoox) [inline frontend-admin/src/app/components/mnoe-staffs-list/mnoe-staffs-list.coffee]: I think the code above is not used anymore.

File: frontend-admin/src/app/index.module.coffee [hunk 5]
```diff
@@ -9,6 +9,8 @@
   'toastr',
   'ncy-angular-breadcrumb',
   'platanus.inflector',
+  'smart-table',
   'angularMoment',
-  'duScroll'
+  'duScroll',
+  'angular.filter',
```
Comment (alexnoox) [inline frontend-admin/src/app/index.module.coffee]: What filters from this lib do you use?
Cool lib ;)

File: frontend-admin/src/app/stylesheets/smart-table.less [hunk 6]
```diff
@@ -0,0 +1,19 @@
+[st-table]{
```
Comment (alexnoox) [inline frontend-admin/src/app/stylesheets/smart-table.less]: 👍

**Comment by alexnoox** [comment]
I'm not sure `frontend-admin/src/app/views/staff/tmp.txt` will be very useful :p

**Comment by xaun** [comment]
when @alexnoox is happy with it I am too @clemthenem, rest is looking good 👍

**Comment by alexnoox** [comment]
@ouranos The travis build is failing but i think you fixed it on your repo. Can we merge this?

**Comment by alexnoox** [comment]
@ouranos After @clemthenem have fixed some stuff of course ;)

**Comment by clemthenem** [comment]
I think I fixed the stuff :) let me know if there Is anything else I need to change !

**Comment by alexnoox** [comment]
@clemthenem Perfect!

@ouranos This can be merged

**Comment by ouranos** [comment]
Yup I need to fix the travis build, I was experimenting with it.
Merging.

**Comment by ouranos** [comment]
@alexnoox could you update the mno-enterprise-demo project with the new build of the admin platform?
It'll be deployed automatically once you push.

**Pull request merged**